### PR TITLE
Add option for timers under pokemon

### PR DESCRIPTION
**Pull request by thomasfa18**
Adds the option to the nav sidebar to display timers under pokemon
Has option in config to enable/disable and set max zoom limit at which timers are present by default in config along with enabling at all.
Rebased and tested off current dev branch

image
image

### CODE REVIEW

**Comment by hammydown4325** [comment]
- Works but labels are off center on copyright safe sprites at certain zooms. 
- I know you played off the functionality from the already existing countdown timer would be nice to remove the parenthesis from the label.
- Also would change the color of the label to not have transparency as its hard to read one on top of the other.
- Also some z-index issues with labels and such that need to get worked out.
- Also make the label.js file conditional on load only if $noTimers is false as most people wont want the extra page load.

2018-02-21 22_44_54-pmsf glennmen
2018-02-21 22_44_22-pmsf glennmen
2018-02-21 22_21_13-pmsf glennmen

**Comment by thomasfa18** [comment]
- [x] Works but labels are off center on copyright safe sprites at certain zooms. **Please confirm this, I couldn't fault it on icons.**
- [x] I know you played off the functionality from the already existing countdown timer would be nice to remove the parenthesis from the label.
- [x] Also would change the color of the label to not have transparency as its hard to read one on top of the other.
- [x] Also some z-index issues with labels and such that need to get worked out. Could not reproduce on my map.
- [x] Also make the label.js file conditional on load only if $noTimers is false as most people wont want the extra page load.

**Comment by billyj85** [comment]
screen shot 2018-03-05 at 8 45 05 pm
I get this in console after pulling the latest PR.

**Comment by thomasfa18** [comment]
@billyj85 after running `npm run build` and cache refresh on browser?

**Comment by billyj85** [comment]
Yep ran npm run build and tried in a couple of different browsers, and one that had never been to the URL before.

**Comment by thomasfa18** [comment]
did you update your config to set timers to true/false as per config.example.php? ie.
```
$noTimers = false;                                                  // true/false
$enableTimers = 'false';                                            // true/false
$hideTimersAtZoomLevel = 16;                                        // true/false
```

**Comment by billyj85** [comment]
Yes, i had it working before the latest update. I tried with both False and True and still had the same results. It also said cant find variable  'google' which all i can think of is the direction provider.

**Comment by billyj85** [comment]
Still can't seem to get it working. Same errors started from fresh clone of Origin Develop.

**Comment by thomasfa18** [comment]
pull the latest commit, I think I must have missmerged (or forgot to pull) with the notification bounce update.
This is working from a fresh git checkout for me

**Comment by billyj85** [comment]
That's got it going, thank you!

**Comment by thomasfa18** [comment]
cool. can you give some feedback on use particular around @iroken222 comments above

**Comment by thomasfa18** [comment]
image
as far as the z indexing goes I still have been unable to fault it. does it only become an issue in a densely populate map @iroken222 ?

**Comment by billyj85** [comment]
@thomasfa18  I haven't had any feedback on timers being misaligned ( I could recreate it if I set icons to XXL size)  When a timer hits 0 it goes  -1min 1sec then disappear after s few more seconds. It's been well received by the few hundred users of the map.

**Comment by hammydown4325** [comment]
I think he fixed the label alignment in a previous commit. By making the timers solid I think the zindex thing probably would be fixed.

**Comment by thomasfa18** [comment]
@billyj85 I've fixed the negative time issues when it ends in this last commit. unless there are any other issues will suggest that this gets merged @Glennmen

**Comment by billyj85** [comment]
Would love to see this merged, it's been working great for the last 10 days with no issues reported by users.

File: static/js/map.js [hunk 1]
```diff
@@ -391,6 +392,8 @@ function initSidebar() {
     $('#pokestops-switch').prop('checked', Store.get('showPokestops'))
     $('#lured-pokestops-only-switch').val(Store.get('showLuredPokestopsOnly'))
     $('#lured-pokestops-only-wrapper').toggle(Store.get('showPokestops'))
+    $('#timer-switch').prop('checked', Store.get('showTimers'))
+    $('#hide-timers-at-zoom-level').val(Store.get('hideTimersAtZoomLevel'))
```
Comment (Glennmen) [inline static/js/map.js:396]: I don't see this element anywhere in pre-index.php, what use does this have?

**Pull request closed without merging**